DeepLabCut/deeplabcut_main.py

```diff
@@ -0,0 +1,180 @@
+
+    # Create project and train network
+    video_path, config_file_path = create_project(filepath, datapath, project_folder, bodypart_names, extracted_frames, deeplabcut_params.params)
+    deeplabcut.create_training_dataset(config_file_path)
+    deeplabcut.train_network(config_file_path)
+    deeplabcut.evaluate_network(config_file_path)
+ 
+    # Analyze video and save the result
+    deeplabcut.analyze_videos(config_file_path, [video_path], destfolder = os.path.dirname(config_file_path))
+    save_coords_to_doric(filepath, datapath, os.path.dirname(config_file_path), deeplabcut_params.params,
+                         group_names = deeplabcut_params.get_h5path_names()) 
+
+
+def preview(deeplabcut_params: dlc_params.DeepLabCutParameters):
+    print("hello preview")
+
+
+def create_project(filepath, datapath, project_folder, bodypart_names, extracted_frames, params):
+    """
+    Create DeepLabCut project folder with config file and labeled data
+    """
+
+    task = os.path.splitext(os.path.basename(filepath))[0]
+    user = "danse"
+
+    with h5py.File(filepath, 'r') as file_:
+        relative_path = file_[datapath].attrs[dlc_defs.Parameters.danse.RELATIVE_FILEPATH]
+
+    full_path = os.path.join(os.path.dirname(filepath), relative_path.lstrip('/'))
+
+    config_file_path = deeplabcut.create_new_project(task, user, [full_path], project_folder, copy_videos = False)
+    
+    update_config_file(config_file_path, bodypart_names)
+
+    create_labeled_data(config_file_path, extracted_frames, bodypart_names, user, params, full_path)
+
+    return full_path, config_file_path
+
+
+def update_config_file(config_file_path, bodypart_names):
+    """
+    Update label names in the config file
+    """
+
+    with open(config_file_path, 'r') as file:
+        data = yaml.safe_load(file)
+  
+    data['bodyparts'] = bodypart_names
+
+    with open(config_file_path, 'w') as file:
+        yaml.safe_dump(data, file, default_flow_style=False)
+
+
+def create_labeled_data(config_file_path, extracted_frames, bodypart_names, experimenter, params, video_path):
+    """
+    Create labeled data in DeepLabCut format
+    """
+
+    # Create folder for labeled data
+    video_name = os.path.splitext(os.path.basename(video_path))[0]
+    dir_path   = os.path.dirname(config_file_path)
+
+    labeled_data_path = os.path.join(dir_path, "labeled-data", video_name)
+    if not os.path.exists(labeled_datapath):
+        os.makedirs(labeled_datapath)
+
+    # Create pandas dataframe with body part coordinates
+    data = []
+    for i in range(len(extracted_frames)):
+        data.append([params[bodypart_name+dlc_defs.Parameters.danse.COORDINATES][i] for bodypart_name in bodypart_names])
+
+    indecies = [[('labeled-data', video_name, f'img{frame}.png')] for frame in extracted_frames]
+    header1  = [[(experimenter, bodypart_name, 'x'), (experimenter, bodypart_name, 'y')] for bodypart_name in bodypart_names]
```

Comment:
Yes, you're right the only way of doing it:
```
header1 = []
for bodypart_name in bodypart_names:
    header1 += [(experimenter, bodypart_name, 'x'), (experimenter, bodypart_name, 'y')]
```
One suggestion `.extend()` and `+=` doing the same thing, so I would pick one, and use everywhere in the code